SentinelHub/tests/test_utils_geo.py

```diff
@@ -0,0 +1,69 @@
+# coding=utf-8
+from typing import Tuple, TypeVar
+
+import pytest
+
+pytest.importorskip("qgis.core")
+
+from qgis.core import QgsRectangle  # noqa: E402
+
+from ..settings import Settings  # noqa: E402
+from ..utils.geo import bbox_to_string, get_custom_bbox, is_bbox_too_large, is_supported_crs  # noqa: E402
+
+T = TypeVar("T", float, int)
+
+
+@pytest.mark.parametrize(
+    "lat_max, lat_min, lng_max, lng_min, coords",
+    [
+        ("1", "0", "1", "0", (0, 0, 1, 1)),
+        ("0", "0", "0", "0", (0, 0, 0, 0)),
+        (
+            "2.0",
+            "1.0",
+            "8.0",
+            "9.0",
+            (8, 1, 9, 2),
+        ),
+    ],
+)
+def test_get_custom_bbox(
+    lat_max: str,
+    lat_min: str,
+    lng_max: str,
+    lng_min: str,
+    coords: Tuple[T, T, T, T],
+) -> None:
+    sett = Settings()
+    sett.lat_max = lat_max
+    sett.lat_min = lat_min
+    sett.lng_max = lng_max
+    sett.lng_min = lng_min
+
+    assert get_custom_bbox(sett) == QgsRectangle(*coords)
+
+
+@pytest.mark.parametrize(
+    "coords, crs, output",
+    [
+        ((10.00300000005, 20.123456789, 23.123456789, 3.123456789), "EPSG:4326", "3.123457,10.003,20.123457,23.123457"),
+        ((10.00300000005, 20.123456789, 23.123456789, 3.123456789), "EPSG:3857", "10.0,3.12,23.12,20.12"),
+        ((1.123456, 2.123456, 3.123456, 4.123456), "EPSG:3535", "1.12,2.12,3.12,4.12"),
+    ],
+)
+def test_bbox_to_string(coords: Tuple[T, T, T, T], crs: str, output: str) -> None:
+    coordinates = bbox_to_string(QgsRectangle(*coords), crs)
+    assert coordinates == output
+
+
+def test_is_bbox_too_large() -> None:
+    assert is_bbox_too_large(QgsRectangle(0, 0, 0, 1), "EPSG:4326", 100000)
+    assert not is_bbox_too_large(QgsRectangle(0, 0, 0, 1), "EPSG:4326", 10000000)
```

Comment:
you might want to use the `1e6` notation, it's hard to see difference between the two lines